core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
What do you think about having only the settings for creating the `KafkaConsumer` as parameter, and we would do the `new KafkaKonsumer` based on those settings? The settings would be immutable, and can be shared across materializations, i.e. no need for a factory.